### ISSUE TITLE
False positive `redundant do` in reify

### ISSUE THREAD

**Issue by serioga** [clj-kondo/clj-kondo#1506]
**version**

clj-kondo v2021.12.16

**macro usage**

Occurs in `reify` macro.

**platform**

windows 10

**problem**

“Redundant do” inside `reify` implementation code.

**repro**

```clojure
(ns clj-kondo-redundant-do
  (:import (clojure.lang IDeref)))

(def ^:private a (atom 0))

(do (reset! a 1)
    nil)

(reify IDeref
  (deref [_] (do (reset! a 1)
                 nil)))
```

```clj_kondo_redundant_do.clj:10:14: warning: redundant do```

**expected behavior**

No warning is expected.

**Comment by borkdude** [comment]
This can be written as:

``` clojure
(reify IDeref
  (deref [_]
    (reset! a 1)
    nil))
```

which is why clj-kondo suggests it's a redundant do.
